File: fe/src/types/comment.ts
import { User } from './user';

// 评论信息
export interface Comment {
  id: number;
  user_id: number;
  note_id: number;
  parent_id?: number;
  content: string;
  like_count: number;
  is_liked?: boolean;
  user?: User;
  replies?: Comment[];
  reply_count?: number;
  created_at?: string;
}

// 创建评论请求
export interface CreateCommentRequest {
  note_id: number;
  content: string;
  parent_id?: number;
  reply_to_user_id?: number;
}

// 评论列表查询参数
export interface CommentListParams {
  note_id: number;
  page?: number;
  page_size?: number;
  type?: 'hot' | 'latest';
}

File: be/internal/model/comment.go
package model

import (
	"gorm.io/gorm"
	"time"
)

type Comment struct {
	ID        uint           `gorm:"primaryKey" json:"id"`
	CreatedAt time.Time      `json:"created_at"`
	UpdatedAt time.Time      `json:"updated_at"`
	DeletedAt gorm.DeletedAt `gorm:"index" json:"deleted_at,omitempty"`

	Content  string `gorm:"size:500;not null" json:"content"`
	NoteID   uint   `gorm:"index;not null" json:"note_id"`
	UserID   uint   `gorm:"index;not null" json:"user_id"`
	ParentID *uint  `gorm:"index;default:0" json:"parent_id"`

	// 【新增】评论的点赞数
	LikeCount uint `gorm:"default:0" json:"like_count"`
	// 回复数（一级评论特有）
	ReplyCount uint `gorm:"default:0" json:"reply_count"`

	User   User     `gorm:"foreignKey:UserID" json:"user,omitempty"`
	Parent *Comment `gorm:"foreignKey:ParentID" json:"parent,omitempty"`
}
